### PR TITLE
test: port test suite to pytest

### PR DESCRIPTION
**Pull request by sliwowitz**
## Summary
- port all test files included in this PR from `unittest` style to pytest-native tests
- replace `unittest.TestCase`, `self.assert*`, and `subTest()` patterns with plain pytest tests, fixtures, helpers, and parametrization
- do a full follow-up DRY pass across the migrated suite to extract shared helpers, remove repetition, and lower test LoC without making the assertions harder to read

## Scope
This PR covers **all test files included in the diff**, across:
- `tests/lib/`
- `tests/cli/`
- `tests/gate/`
- `tests/scripts/`
- `tests/tui/`
- shared test helpers under `tests/`

## Validation
- `make lint`
- `make check`

Latest full validation result:
- full suite passed: `1204 passed`
- `tach check` passed
- docstring coverage passed
- `vulture` passed
- `reuse lint` passed

## Summary by CodeRabbit

* **Tests**
  * Large-scale modernization: migrated many suites from unittest to pytest with fixtures, parametrization, and module-level helpers.
  * Broadened coverage across CLI, library, gate/server, container, image, workflow, and project areas with more data-driven and edge-case scenarios.
  * Added numerous reusable test utilities, fixtures, constants, dataclasses, and lightweight helpers to simplify setup, assertions, and CLI/task harnessing.

### CODE REVIEW

File: tests/lib/test_images.py [hunk 1]
```diff
@@ -1,161 +1,134 @@
-# SPDX-FileCopyrightText: 2025 Jiri Vyskocil
+# SPDX-FileCopyrightText: 2026 Jiri Vyskocil
```
[the remark this answers is not in the document]
Comment (sliwowitz) [reply tests/lib/test_images.py]: I checked the file history before changing this. `tests/lib/test_images.py` was added on 2026-01-08 in commit `9c02fff`, so `2026` is already the first-contribution year. Keeping the header as-is.

File: tests/lib/test_task_names.py [hunk 2]
```diff
@@ -1,13 +1,16 @@
 # SPDX-FileCopyrightText: 2026 Jiri Vyskocil
```
[the remark this answers is not in the document]
Comment (sliwowitz) [reply tests/lib/test_task_names.py]: I checked the file history before changing this. `tests/lib/test_task_names.py` was added on 2026-03-01 in commit `980eba8`, so `2026` is already the first-contribution year. Keeping the header as-is.
